src/nat/libp2p_nat_server.erl

```diff
@@ -114,9 +114,11 @@ handle_info(renew, #state{tid=TID, transport_tcp=Pid, internal_address=IntAddr,
             end,
             {noreply, State#state{external_address=ExtAddr1, external_port=ExtPort1, lease=Lease, since=Since}};
         {error, _Reason} ->
+            %% this might lead to a period of undialability
             ok = remove_multi_addr(TID, ExtAddr0, ExtPort0),
             lager:warning("failed to renew lease for port ~p: ~p", [{IntPort, ExtPort0}, _Reason]),
-            {stop, renew_failed}
+            _ = erlang:send_after(5000, self(), renew),
```

Comment:
this is likely far too short, would suggest at least 5 minutes AND telling the transport_tcp to set `negotiated_nat = false` in the meantime.